src/Apps/Order/Components/AddressVerificationManager.tsx

```diff
@@ -0,0 +1,357 @@
+import {
+  ModalDialog,
+  Spacer,
+  Box,
+  Flex,
+  Button,
+  RadioGroup,
+  BorderedRadio,
+  Text,
+} from "@artsy/palette"
+import { useSystemContext } from "System/SystemContext"
+import { FC, useState, useEffect } from "react"
+import { fetchQuery, Environment, graphql } from "react-relay"
+import {
+  AddressVerificationManagerQuery,
+  AddressVerificationManagerQuery$data,
+} from "__generated__/AddressVerificationManagerQuery.graphql"
+
+type PresentVerificationPayload = NonNullable<
+  AddressVerificationManagerQuery$data["verifyAddress"]
+>
+type SuggestedAddresses = PresentVerificationPayload["suggestedAddresses"]
+
+type InputAddress = PresentVerificationPayload["inputAddress"]
+
+interface VerifiableAddress {
+  addressLine1: string
+  addressLine2?: string
+  country: string
+  city: string
+  region: string
+  postalCode: string
+}
+
+type IncomingAddressValues = VerifiableAddress & {
+  [key: string]: unknown
+}
+
+type AddressOption = {
+  address: VerifiableAddress
+  lines: string[]
+  recommended: boolean
+  key: string
+}
+
+enum ErrorModalTitle {
+  general = "Check your delivery address",
+  suggested = "Confirm your delivery address",
+}
+
+enum AddressSuggestionRadioButton {
+  recommended = "Recommended",
+  user_address = "What you entered",
+}
+
+export const AddressVerificationManager: FC<{
+  render: React.FC<{
+    verifyAddress: (address: VerifiableAddress) => React.ReactNode
+    resetModal: () => void
+  }>
+}> = ({ render }) => {
```

Comment:
:ceiling_cat: What's the upside of using a function param? Is this possible by passing something like `({ children })` instead?

If it is, then it could simplify some of the code below if I noted.

Reply:
The reason (using this approach) was that the children/render component needs to accept a prop created in this components body, so it can only be passed in after that is created. However, Damon pointed out a much more elegant solution to me using a `QueryRenderer`.